modules/ROOT/pages/_partials/partner-and-customer-guide.adoc

```diff
@@ -130,10 +132,29 @@ See the https://docs.oracle.com/en/java/javase/17/migrate/getting-started.html#G
 See xref:java-support.adoc#mule-runtime[Mule runtime] to learn more about the Java 17 changes made in Mule runtime engine. 
 +
 ** Reflection on Java classes
++
+To ensure compatibility with Java 17, review and update any code that uses reflection to access Java classes or set parameter values. Java 17 imposes stricter access controls, so you should avoid introspecting JDK internals. Modify your code to eliminate deep introspection of JDK classes and ensure that reflective access is limited to necessary areas only.
 ** Serialization
++
+Update your serialization code to accommodate the restrictions of Java 17. If your code uses libraries like Gson for serializing classes such as MetadataType, switch to using Data Transfer Objects (DTOs) where reflective access to JDK classes is involved. Ensure that serialization libraries are updated and adjust your code to minimize reliance on reflection.
```

Comment:
`such as MetadataType` -> This is an internal class, most customers don't know about it and definitvely not one they would be serializing themselves